customfields.go

```diff
@@ -0,0 +1,114 @@
+package katana
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/projectdiscovery/fileutil"
+	"github.com/projectdiscovery/gologger"
+	errorutil "github.com/projectdiscovery/utils/errors"
+
+	"github.com/projectdiscovery/katana/pkg/utils"
+	"gopkg.in/yaml.v2"
+)
+
+//go:embed field-config.yaml
+var FieldConfig []byte
+
+var AllCustomFieldsMap = make(map[string]CustomFieldConfig)
+
+// CustomFieldConfig contains suggestions for field filling
+type CustomFieldConfig struct {
+	Name         string           `yaml:"name,omitempty"`
+	Type         string           `yaml:"type,omitempty"`
+	Part         string           `yaml:"part,omitempty"`
+	Group        int              `yaml:"group,omitempty"`
+	Regex        []string         `yaml:"regex,omitempty"`
+	CompileRegex []*regexp.Regexp `yaml:"-"`
+}
+
+type Part string
+
+const (
+	// RequestPart is the part of request
+	Header   Part = "header"
+	Body     Part = "body"
+	Response Part = "response"
+)
+
```

Comment:
@ShubhamRasal I think we can covert this to a proper enum type and since this is not directly needed and used in other files in katana we should more `Part` to utils.go